crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -1691,43 +1691,34 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.infer_annotation_expression(&type_alias.value, DeferredExpressionState::Deferred);
     }
 
-    /// Returns `true` if the current scope is the function body scope of a method of a protocol
-    /// (that is, a class which directly inherits `typing.Protocol`.)
-    fn in_protocol_class(&self) -> bool {
+    /// If the current scope is a method inside an enclosing class,
+    /// return `Some(class)` where `class` represents the enclosing class.
+    ///
+    /// If the current scope is not a method inside an enclosing class,
+    /// return `None`.
+    fn class_context_of_current_method(&self) -> Option<ClassLiteral<'db>> {
```

Comment:
We already have a function `enclosing_class_symbol` (which should probably be renamed because it returns a Type,  not a Symbol) -- seems like there's likely redundancy between this method and that one?